docs/reference/remap/functions/redact.cue

```diff
@@ -0,0 +1,94 @@
+package metadata
+
+remap: functions: redact: {
+	category: "String"
+	description: """
+		Redact potentially sensitive data in `value` such as:
+
+		- credit card numbers
+		- custom patterns
+		- (more to come!)
+
+		This can act as a last resort in the case your applications log sensitive data by accident to ensure you do not
+		leak this information into your log storage.
+		"""
+
+	arguments: [
+		{
+			name: "value"
+			description: #"""
+				The value to redact sensitive data from.
+
+				Its behavior differs depending on the type of `value`:
+				- For strings, it simply redacts the sensitive data and returns a new string
+				- For arrays, it redacts the sensitive data in each string element
+				- For objects, it masks the sensitive data in each string value, but not keys
+
+				For arrays and objects it will recurse into any nested arrays or objects. Any non-string elements will
+				be skipped.
+
+				Any redacted text will be replaced with `[REDACTED]`.
+				"""#
+			required: true
+			type: ["string", "object", "array"]
+		},
+		{
+			name: "filters"
+			description: #"""
+					List of filters to be applied to the `value`.
+
+					If none, the default set of filters is applied: ["credit_card"].
+
+					Each filter can be specified in one of three ways:
+					- As a regular expression directly, which will be used to redact text matching it
+					- As an object with a `type` key that corresponds to a named filter and additional keys for customizing that filter
+					- As a named filter, if it has no required parameters
+
+					Named filters are:
+					- `pattern`: Redact text matching any regular expressions specified in the, required, `patterns`
+					   key. This is the expanded form of just passing a regular expression as a filter.
+					- `credit_card`: Redact credit card numbers.
```

Comment:
Shouldn't this be represented via an `enum` attribute like other enumerations?

Reply:
Hmm, true, though this one is a bit different in that the argument can be an enum, a regex, or an object rather than just an enum; which is a bit unique as far as the VRL function parameters go. I don't think the cue data model is really set up to model that at the moment. Its type is just `array`.